q2_phylogeny/_filter.py

```diff
@@ -19,3 +19,19 @@ def filter_table(table: biom.Table, tree: skbio.TreeNode) -> biom.Table:
     ids_to_keep = tip_ids & feature_ids
     table.filter(ids_to_keep, axis='observation', inplace=True)
     return table
+
+
+def filter_tree(table: biom.Table, tree: skbio.TreeNode) -> skbio.TreeNode:
+    """
+    Prunes a phylogenetic tree to match the input feature table
+    """
+    tip_ids = set([t.name for t in tree.tips()])
+    feature_ids = set(table.ids(axis='observation'))
+    # ids_to_keep can only include ids that are in table
+    ids_to_keep = tip_ids & feature_ids
```

Comment:
Would it be better to be more explicit and simply check if the table is a subset of the tree first? I wonder if this logic would trip-up users up if they are incorrectly mixing trees and tables (say after various iterations of their data).

I feel that checking if the table is a subset, would fit the help text more directly, *i.e.*  "Remove features from tree if they're not present in table."  But that does not appear to actually be the case, we're just keeping the intersection. Is this correct?

That is, you can have a feature in the table that is not in the tree. Which leads me to ask, why would this happen unless other filtering was performed? Perhaps raise a user warning that the table is not a subset of the tree?

Perhaps I'm not thinking of other use cases for this? 🤔

Reply:
Yes, that's probably much smarter!

Reply:
To be honest, I am not sure. I am leaning towards an error... unless there is a use case I've not thought of. 

I imagine this is intended for the cases in which a tree is made from an initial large feature-table table... then downstream, the feature-table is filtered. Instead of remaking a new tree from scratch (based on this new filtered table), you simply subset the full tree to match the newly filtered feature-table?